### ISSUE TITLE
Feedback homework week 2

### ISSUE THREAD

**Issue by remarcmij** [ThreePalmTrees/hyf-javascript1#1]
Hi Husam, here is my feedback on your homework for week 2.

I think you did a pretty good job, but make sure you run your code again whenever you make a change.

### 1-strings.js

This file doesn't run. It produces this error message:

```
easyToRead = myString.replace(/,/g, " ");
           ^

ReferenceError: easyToRead is not defined
```

I think you added a `'use strict';` line afterwards and forgot to run this file again. The `'use strict';` line prevents you from using undeclared variables (it protects against making errors like this: a good thing). You have not declared the variable `easyToRead` with a `const` or `let` keyword.

### 2-arrays.js

Perfect!

### 3-months.js

Perfect!

## 4-maartje.js

Here I think the same thing happened as with `1-strings.js`: you added `'use strict';` without running your code again to check its effect. The problem is that you didn't declare the loop index variable `i`. You need to declare it with a `let` like this:

```js
for (let i = 0; i < tasks.length; i++) {
  //...
}
```

It is best to declare variables just before the point you need them. Also use `const` if you do not intend the modify the variables:

```js
const hourSalary = 23;
const earnings = hoursWorked * hourSalary;
console.log("Maartje has earned: " + earnings.toFixed(2) + " euros.");
```

**Comment by remarcmij** [comment]
>  ...hourSalary I kept using `let`...

You are absolutely right. For the same reason I choose my age to be a `let` during the lectures 😅.

**Comment by ThreePalmTrees** [comment]
:smile:
